index.js

```diff
@@ -36,32 +36,59 @@ async function addRouteThrottleHook (fastify, routeOptions, throttleOptions) {
 function throttleOnSendHandler (fastify, throttleOpts) {
   const bytesPerSecond = throttleOpts.bytesPerSecond
 
-  return function onSendHandler (request, reply, payload, done) {
-    if (throttleOpts.streamPayloads && payload instanceof Stream) {
-      done(null, pipeline(
-        payload,
-        new ThrottleStream({ bytesPerSecond }),
-        err => { fastify.log.error(err) }
-      ))
-      return
+  if (typeof bytesPerSecond === 'number') {
```

Comment:
I would keep the async style only - much more easier to read. 
Or the performance would be awful at that point? 😄

Reply:
You mean check if it is an async function and then only await?

Reply:
use always `async onSendHandler` and await `bytesPerSecond` if necessary

Reply:
then I have to detect if it is an async function. Well I will probably restrict it to only async function but not functions returning Promises.